src/os/bluestore/KernelDevice.cc

```diff
@@ -548,3 +557,22 @@ int KernelDevice::invalidate_cache(uint64_t off, uint64_t len)
   return r;
 }
 
+int KernelDevice::aio_read(uint64_t off, uint64_t len, bufferlist *pbl, IOContext *ioc, bool buffered)
```

Comment:
@liewegas . From this(http://lse.sourceforge.net/io/aio.html)
What Works?
>>AIO read and write on raw (and O_DIRECT on blockdev)
>>AIO read and write on files opened with O_DIRECT on ext2, ext3, jfs, xfs
What Does Not Work?
>>AIO read and write on files opened without O_DIRECT (i.e. normal buffered filesystem >>AIO). On ext2, ext3, jfs, xfs and nfs, these do not return an explicit error, but quietly default >>to synchronous or rather non-AIO behaviour (i.e io_submit waits for I/O to complete in these cases). For most other filesystems, -EINVAL is reported.

From the "What works". it need open w/ O_DIRECT for raw block.
But for "What Does Not Work",, it don't say for raw block w/o O_DIRECT.
Have you time to explain in detail? Thanks!